cylc/flow/workflow_files.py

```diff
@@ -1745,53 +1746,30 @@ def detect_flow_exists(
 
 def check_flow_file(
     path: Union[Path, str],
-    symlink_suiterc: bool = False,
-    logger: Optional['Logger'] = LOG
 ) -> Path:
-    """Raises WorkflowFilesError if no flow file in path sent.
+    """Checks the path for a suite.rc or flow.cylc file.
+
+    Raises:
+        WorkflowFilesError
+            - if no flow file in path sent
+            - both suite.rc and flow.cylc in path sent.
 
     Args:
         path: Absolute path to check for a flow.cylc and/or suite.rc file.
-        symlink_suiterc: If True and suite.rc exists, create flow.cylc as a
-            symlink to suite.rc. If a flow.cylc symlink already exists but
-            points elsewhere, it will be replaced.
-        logger: A custom logger to use to log warnings.
 
     Returns the path of the flow file if present.
     """
     flow_file_path = Path(expand_path(path), WorkflowFiles.FLOW_FILE)
     suite_rc_path = Path(expand_path(path), WorkflowFiles.SUITE_RC)
     if flow_file_path.is_file():
-        if not flow_file_path.is_symlink():
-            if not suite_rc_path.is_file():
-                return flow_file_path
-            raise WorkflowFilesError(
-                f"Both {WorkflowFiles.FLOW_FILE} and "
-                f"{WorkflowFiles.SUITE_RC} files are present in the source "
-                "directory. Please remove one and try again. For more "
-                "information visit: https://cylc.github.io/cylc-doc/latest/"
-                "html/7-to-8/summary.html#backward-compatibility"
-            )
-        if flow_file_path.resolve() == suite_rc_path.resolve():
-            # A symlink that points to existing suite.rc
-            return flow_file_path
-    if suite_rc_path.is_file():
-        if not symlink_suiterc:
-            return suite_rc_path
-        if flow_file_path.is_symlink():
-            # Symlink broken or points elsewhere - replace
-            flow_file_path.unlink()
-        flow_file_path.symlink_to(WorkflowFiles.SUITE_RC)
-        if logger:
-            logger.warning(
-                f"Symlink created: "
-                f"{WorkflowFiles.FLOW_FILE} -> {WorkflowFiles.SUITE_RC}"
-            )
+        detect_both_flow_and_suite(Path(path))
         return flow_file_path
+    if suite_rc_path.is_file():
+        return suite_rc_path
     raise WorkflowFilesError(NO_FLOW_FILE_MSG.format(path))
 
 
-def create_workflow_srv_dir(rundir=None, source=None):
+def create_workflow_srv_dir(rundir=None):
```

Comment:
This would not pass Mypy
```suggestion
def create_workflow_srv_dir(rundir: Path) -> None:
```